src/website/shared/auth.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from typing import Any, cast
+
+from allauth.socialaccount.models import SocialAccount
+from django.contrib.auth.models import Group, Permission, User
+from django.db.models import Q
+from github import Github
+from github.NamedUser import NamedUser
+from github.Organization import Organization
+from github.Team import Team
+
+from shared.utils import get_gh
+
+github: Github = get_gh(per_page=100)  # 100 is the API limit
+logger = logging.getLogger(__name__)
+
+
+def get_gh_username(user: User) -> str | None:
+    """
+    Return the Github username of a given Auth.User.
+    """
+    social_user = User.objects.get(id=user.id)  # type: ignore
+    social_account: SocialAccount | None = (
+        social_user.socialaccount_set.filter(provider="github").first()  # type: ignore
+    )
+    if social_account:
+        return social_account.extra_data.get("login")  # type: ignore
+
+    logger.warning(f"Failed to get GitHub username for user {user}.")
+    return None
+
+
+def get_gh_organization(orgname: str) -> Organization | None:
+    """
+    Return the Github Organization instance given an organization name.
+    """
+    try:
+        return github.get_organization(login=orgname)
+    except Exception as e:
+        logger.warning(f"Failed to get organization {orgname}: {e}")
```

Comment:
`logger.exception` should probably be used here.

Reply:
Done [c698428](https://github.com/Nix-Security-WG/nix-security-tracker/pull/159/commits/c69842824f83be36e467f0d997f0afead9e0e01f)